10-angular/ui_router/readme.md

```diff
@@ -0,0 +1,389 @@
+
+**I'm going to remove two Angular libraries** out of the three included on this page, just by deleting their `<script>` tags. Now I only have two modules: `ng` and `ngLocale`. These are the only two that come out-of-the-box with Angular.
+
+![Two default modules](http://i.imgur.com/dRgcpXi.png)
+
+## Here there be dragons
+
+Now, only those of you in Chrome again, open the last `Closure` inside `<function scope>`. You should get something terrifying-looking: a bunch of important-looking methods that are all named completely unhelpful things like `a`, `b`, `c`, and so on.
+
+![Letter-named methods](http://i.imgur.com/yIxpbnA.png)
+
+Q. What is this stuff? (Hint: We're loading Angular from a file called `angular.min.js`.
+---
+> This is minified code. When you minify Javascript, in order to save memory functions and variables are renamed to just letters of the alphabet. 
+
+If I remove the `.min` from the `<script>` tag in my HTML, all of these indecipherable function names are replaced with English-y function names like `$AnchorScrollProvider`, `MOZ_HACK_REGEXP`, `PRISTINE_CLASS`, and so on.
+
+> They're in a different order because they're listed alphabetically.
+
+![English-named methods](http://i.imgur.com/g9DtO4A.png)
+
+These are all functions and constants used internally by Angular. The effects of minifaction will be brought up again later on in this class.
+
+# Our first module
+
+We're going to create our own Angular module. It's not going to do anything for a bit. For the next part of this class, we're only going to focus on setting up modules.
+
+Notice there's an `app.js` linked in this `index.html`. Open it and add this line:
+
+```js
+angular.module("grumblr", []);
+```
+
+**It's very important that you include that empty array.**
+
+If I go poking around with `angular.module` in the console again, I can see that `grumblr` has been added to Angular's available modules:
+
+![Grumblr in the modules](http://i.imgur.com/s61Z6VP.png)
+
+The `requires` part of `grumblr` is currently an empty array.
+
+Now, modify `app.js` to look like this:
+
+```js
+angular.module("grumblr", ["ui.router"]);
+```
+
+We put something called "ui.router" in that array. You may recognize it as another one of the modules we have. Checking `grumblr.requires` again, it's now an array with one thing inside it -- `ui.router`!
+
+Q. Based on your experiences with the word `require` so far, what does this array in `angular.module` do?
+---
+> It says which dependencies should be included in this module.
+
+## Dependency Injection
+
+The process of requiring dependencies in Angular is called **dependency injection**. It's an extremely important part of Angular since this framework is all about modules being dependent on other modules.
+
+#### Try removing that array altogether. What happens?
```

Comment:
Should we have some talk of scope and IIFE here?  I don't think was highlighted in the first lesson.  re: We need something to manage/control/share scope for our app, since all of this code is wrapped within it's own scope.